src/compas_view2/app/app.py

```diff
@@ -129,33 +133,85 @@ class App:
-        controller_class: Optional[Controller] = None,
-        show_grid: bool = True,
-        config: Optional[dict] = None,
-        enable_sidebar: bool = False,
-        enable_sidedock1: bool = False,
-        enable_sidedock2: bool = False,
-        enable_sceneform: bool = False,
-        enable_propertyform: bool = False,
-        show_flow: bool = False,
+        title: str = None,
+        version: Literal["120", "330"] = None,
+        width: int = None,
+        height: int = None,
+        viewmode: Literal["wireframe", "shaded", "ghosted", "lighted"] = None,
+        show_grid: bool = None,
+        enable_sidebar=None,
+        enable_sidedock1: bool = None,
+        enable_sidedock2: bool = None,
+        enable_sceneform: bool = None,
+        enable_propertyform: bool = None,
+        show_flow: bool = None,
         flow_view_size: Union[Tuple[int], List[int]] = None,
-        flow_auto_update: bool = True,
+        flow_auto_update: bool = None,
+        config: Optional[dict] = None,
+        controller_class: Optional[Controller] = None,
     ):
-        if version not in VERSIONS:
+        # Initialize the config.
+        if config is not None:
+            if not isinstance(config, dict):
+                with open(config) as f:
+                    config = json.load(f)
+
+            for key in DEFAULT_CONFIG:
+                if key not in config:
+                    config[key] = DEFAULT_CONFIG[key]
+        else:
+            config = DEFAULT_CONFIG
+
+        # config variables for the app.
+        if title is not None:
+            config["app"]["title"] = title
+        if version is not None:
+            config["app"]["version"] = version
+        if width is not None:
+            config["app"]["width"] = width
+        if height is not None:
+            config["app"]["height"] = height
+
+        if viewmode is not None:
+            config["view"]["viewmode"] = viewmode
+        if show_grid is not None:
+            config["view"]["show_grid"] = show_grid
+
+        if enable_sidebar is not None:
+            config["sidebar"]["enable_sidebar"] = enable_sidebar
+
+        if enable_sidedock1 is not None:
+            config["sidedocks"]["enable_sidedock1"] = enable_sidedock1
+        if enable_sidedock2 is not None:
+            config["sidedocks"]["enable_sidedock2"] = enable_sidedock2
+        if enable_sceneform is not None:
+            config["sidedocks"]["enable_sceneform"] = enable_sceneform
+        if enable_propertyform is not None:
+            config["sidedocks"]["enable_propertyform"] = enable_propertyform
+
+        if show_flow is not None:
+            config["flow"]["show_flow"] = show_flow
+        if flow_view_size is not None:
+            config["app"]["flow_view_size"] = flow_view_size
+        if flow_auto_update is not None:
+            config["app"]["flow_auto_update"] = flow_auto_update
+
+        self.config = config["app"]
+        self.all_config = config
```

Comment:
I suggest `self.config_app = config["app"]`, then `self.config = config` , what do you think?

Reply:
each class (like `controller`, and `flow` has a unified attribute `.config`), `app` is the same. while since `app` is the entry for all the configs, I created the `.all_config`.  I think we can keep it in this version. Apart from this, the rest comments are fixed.